### PR TITLE
Adds rust project tikv

### PR DESCRIPTION
**Pull request by catenacyber**
@koushiro @breeswish @overvenus

would you be interested in continuous fuzzing for tikv ?
This PR enables it with oss-fuzz with your existing fuzz targets, with libFuzzer (only)
I added a quick and dirty patch to be able to build the fuzz targets without running them at once...

### CODE REVIEW

**Review by breeswish — COMMENTED** [review]
Thanks a lot! It would be very great to fuzz TiKV continuously. I added some comments below.

File: projects/rust-tikv/build.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# limitations under the License.
+#
+################################################################################
+
+git apply $SRC/patch.diff
+cd fuzz
+RUSTFLAGS="" cargo run -- list-targets | while read t; do
+    RUSTFLAGS="" cargo run -- run $FUZZING_ENGINE ${t}
+    cp ../target/x86_64-unknown-linux-gnu/debug/${t} $OUT/
+done
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply projects/rust-tikv/build.sh:23]: Yes, for every fuzz target as listed by `cargo run -- list-targets`

File: projects/rust-tikv/patch.diff [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+@@ -318,9 +318,9 @@ fn run_libfuzzer(target: &str) -> Result<()> {
+     asan_options.push_str(" detect_odr_violation=0");
+ 
+     let fuzzer_bin = Command::new("cargo")
+-        .args(&["run", "--target", &target_platform, "--bin", target, "--"])
+-        .arg(&corpus_dir)
+-        .arg(&seed_dir)
++        .args(&["build", "--target", &target_platform, "--bin", target, "--"])
++        //.arg(&corpus_dir)
++        //.arg(&seed_dir)
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply projects/rust-tikv/patch.diff:14]: I think some changes can be brought to TIKV, but this diff is rather a POC.

I have been working a bit more on this here : 
https://github.com/catenacyber/targets/tree/buildfuzz
See also #5715
Essentially, that means adding a `build` option in addition to the `run` one
Do you want me to make a PR to tikv about it ?

[the remark this answers is not in the document]
Comment (breeswish) [reply projects/rust-tikv/patch.diff:14]: Yes it would be great!

[the remark this answers is not in the document]
Comment (catenacyber) [reply projects/rust-tikv/patch.diff]: Opened https://github.com/tikv/tikv/pull/10123

And updated the PR about this... let's wait for oss-fuzz CI

**Comment by catenacyber** [comment]
Thanks Oliver for cleaning up, I will reopen if tikv shows interest again

**Pull request closed without merging**